src/control/server/config.go

```diff
@@ -389,30 +382,31 @@ func (c *Configuration) Validate() (err error) {
 	defer func() {
 		if err != nil {
 			examplesPath, _ := c.ext.getAbsInstallPath(relConfExamplesPath)
-			err = errors.WithMessage(err, msgBadConfig+examplesPath)
+			err = errors.WithMessage(FaultBadConfig,
+				err.Error()+", examples: "+examplesPath)
 		}
 	}()
 
 	if c.Fabric.Provider == "" {
-		return errors.New(msgConfigNoProvider)
+		return FaultConfigNoProvider
 	}
 
 	// only single access point valid for now
 	if len(c.AccessPoints) != 1 {
-		return errors.New(msgConfigBadAccessPoints)
+		return FaultConfigBadAccessPoints
 	}
 	// apply configured control port if not supplied
 	for i := range c.AccessPoints {
-		if !hasPort(c.AccessPoints[i]) {
+		if !common.HasPort(c.AccessPoints[i]) {
```

Comment:
No need to change, but it just occurred to me that HasPort seems like a good method to have an on Addr type (which would just be a typedef of a string). Maybe a refactoring that could occur in the future.